cuda_core/tests/test_program.py

```diff
@@ -13,6 +13,106 @@
 is_culink_backend = _linker._decide_nvjitlink_or_driver()
 
 
+def _is_nvvm_available():
+    """Check if NVVM is available."""
+    try:
+        from cuda.core.experimental._program import _get_nvvm_module
+
+        _get_nvvm_module()
+        return True
+    except ImportError:
+        return False
+
+
+nvvm_available = pytest.mark.skipif(
+    not _is_nvvm_available(), reason="NVVM not available (libNVVM not found or cuda-bindings < 12.9.0)"
+)
+
+
+@pytest.fixture(scope="session")
+def nvvm_ir():
+    """Generate working NVVM IR with proper version metadata.
+    The try clause here is used for older nvvm modules which
+    might not have an ir_version() method. In which case the
+    fallback assumes no version metadata will be present in
+    the input nvvm ir
+    """
+    try:
+        from cuda.core.experimental._program import _get_nvvm_module
+
+        nvvm = _get_nvvm_module()
+        major, minor, debug_major, debug_minor = nvvm.ir_version()
+
+        nvvm_ir_template = """target triple = "nvptx64-unknown-cuda"
+target datalayout = "e-p:64:64:64-i1:8:8-i8:8:8-i16:16:16-i32:32:32-i64:64:64-i128:128:128-" \
+"f32:32:32-f64:64:64-v16:16:16-v32:32:32-v64:64:64-v128:128:128-n16:32:64"
+
+define i32 @ave(i32 %a, i32 %b) {{
+entry:
+  %add = add nsw i32 %a, %b
+  %div = sdiv i32 %add, 2
+  ret i32 %div
+}}
+
+define void @simple(i32* %data) {{
+entry:
+  %0 = call i32 @llvm.nvvm.read.ptx.sreg.ctaid.x()
+  %1 = call i32 @llvm.nvvm.read.ptx.sreg.ntid.x()
+  %mul = mul i32 %0, %1
+  %2 = call i32 @llvm.nvvm.read.ptx.sreg.tid.x()
+  %add = add i32 %mul, %2
+  %call = call i32 @ave(i32 %add, i32 %add)
+  %idxprom = sext i32 %add to i64
+  store i32 %call, i32* %data, align 4
+  ret void
+}}
+
+declare i32 @llvm.nvvm.read.ptx.sreg.ctaid.x() nounwind readnone
+declare i32 @llvm.nvvm.read.ptx.sreg.ntid.x() nounwind readnone
+declare i32 @llvm.nvvm.read.ptx.sreg.tid.x() nounwind readnone
+
+!nvvm.annotations = !{{!0}}
+!0 = !{{void (i32*)* @simple, !"kernel", i32 1}}
+
+!nvvmir.version = !{{!1}}
+!1 = !{{i32 {major}, i32 0, i32 {debug_major}, i32 0}}
+"""
```

Comment:
@abhilash1910 I was scratching my head as our other NVVM test file also has long lines but the linter does not complain at all. This is the key:
https://github.com/NVIDIA/cuda-python/blob/d64120bb6733391989174456daaf18643ea29711/cuda_bindings/tests/test_nvvm.py#L37
Let's apply this change and remove the unnecessary line breaks in the text IRs.
```suggestion
"""  # noqa: E501
```

Reply:
This is great and most easiest fix, I tried f strings and a lot of other hacks but that always increases the code lines. 
Thanks @leofang